dali/python/nvidia/dali/experimental/torchvision/v2/compose.py

```diff
@@ -0,0 +1,197 @@
+# Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Sequence, Union, Callable
+
+from nvidia.dali.pipeline import Pipeline
+from nvidia.dali import _conditionals
+from nvidia.dali._utils.dali_trace import set_tracing
+from nvidia.dali.data_node import DataNode as _DataNode
+import nvidia.dali.ops as ops
+from nvidia.dali.backend import TensorListCPU, TensorListGPU
+
+from .tensor import ToTensor
+
+from PIL import Image
+import numpy as np
+import torch
+
+
+def _to_torch_tensor(tensor_or_tl: Union[TensorListGPU, TensorListCPU]) -> torch.Tensor:
+    if isinstance(tensor_or_tl, (TensorListGPU, TensorListCPU)):
+        dali_tensor = tensor_or_tl.as_tensor()
+    else:
+        dali_tensor = tensor_or_tl
+
+    return torch.from_dlpack(dali_tensor)
+
+
+def to_torch_tensor(tensor_or_tl: Union[tuple, TensorListGPU, TensorListCPU]) -> torch.Tensor:
+
+    if isinstance(tensor_or_tl, tuple) and len(tensor_or_tl) > 1:
+        tl = []
+        for elem in tensor_or_tl:
+            tl.append(_to_torch_tensor(elem))
+        return tuple(tl)
+    else:
+        if len(tensor_or_tl) == 1:
+            tensor_or_tl = tensor_or_tl[0]
+        return _to_torch_tensor(tensor_or_tl)
+
+
+class Compose(Pipeline):
+    """
+    Compose several DALI transforms together.
+
+    This class inherits from nvidia.dali.Pipeline and allows chaining multiple
+    DALI operations in a sequential manner, similar to torchvision.transforms.Compose.
+    The Compose runs implements a callable which runs a pipeline.
+
+    """
+
+    def __init__(
+        self,
+        op_list: List[Callable[..., Union[Sequence[_DataNode], _DataNode]]],
+        batch_size: int = 1,
+        num_threads: int = 1,
+        *args,
+        **kwargs,
```

Comment:
Those `args` and `kwargs` are the arguments I'd pass to DALI pipeline? If so it might be good to provide this info in documentation. Maybe even call it such? `**dali_pipeline_args`.

Also, I'm not sure if having `*args` after defined keyword arguments is a good pattern. Do we have any specific usages for this? Maybe we can just omit `*args` and have keyword args only?